### ISSUE TITLE
linter: Add option to require a description for oxlint disable directives

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/oxc#22193]
With eslint there is a [plugin](https://www.npmjs.com/package/@eslint-community/eslint-plugin-eslint-comments) that adds a rule to require a description when disabling an eslint rule through a comment directive. It would be great if this was just built into oxlint as an option.

**Suggestion**
New option: `reportUndescribedDirectives`

With the new option enabled these would fail:
```
// oxlint-disable
// oxlint-disable-next-line no-console
/* oxlint-disable no-alert */
```

These would be allowed:
```
// oxlint-disable -- Special file, no need to lint
// oxlint-disable-next-line no-console -- CLI usage, console is ok
/* oxlint-disable no-alert -- Debug file */
```

**Comment by eai04191** [comment]
Just to add a data point in support of native implementation: `@eslint-community/eslint-plugin-eslint-comments` can't fill this gap for `oxlint-disable-*` directives.

The `require-description` rule's directive detection is hardcoded to the `eslint-` prefix via this regex in `lib/internal/utils.js`:

```js
const DIRECTIVE_PATTERN =
    /^(eslint(?:-env|-enable|-disable(?:(?:-next)?-line)?)?|exported|globals?)(?:\s|$)/u
```
https://github.com/eslint-community/eslint-plugin-eslint-comments/blob/de049807e525ac3e2d164c8873f9035b08eb23a6/lib/internal/utils.js#L10-L11

So even when wired up through oxlint's JS plugin bridge, it silently ignores every `// oxlint-disable-next-line ...` comment and only flags the eslint- form. Projects that have standardized on the oxlint- prefix have no path forward without native support. hence the need for this option in oxlint itself.

**Comment by camc314** [comment]
Hi, thanks for making issue.

I'm closing this as not planned for now, as it's not a high priority.

If you want to report on these missing descriptions, please use a custom plugin. this is <30 LOC.

```ts
// plugin.ts
import type { Plugin } from "oxlint/plugins";

const plugin: Plugin = {
  meta: {
    name: "my-plugin",
  },
  rules: {
    "require-disable-directive-description": {
      create(context) {
        return {
          Program(_program) {
            context.sourceCode.getDisableDirectives().directives.forEach((directive) => {
              if (!directive.justification) {
                context.report({
                  node: directive.node,
                  message: "Disable directive should have a description.",
                });
              }
            });
          }
        };
      },
    },
  },
};

export default plugin;

```

```json
// .oxlintrc.json
{
  "jsPlugins": ["./plugin.ts"],
  "rules": {
    "my-plugin/require-disable-directive-description": "error"
  }
}
```


Output when running the linter:
```
  x my-plugin(require-disable-directive-description): Disable directive should have a description.
   ,-[files/index.js:1:1]
 1 | // oxlint-disable-next-line
   : ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 2 | debugger;
   `----

  x my-plugin(require-disable-directive-description): Disable directive should have a description.
   ,-[files/index.js:3:1]
 2 | debugger;
 3 | // oxlint-disable-next-line no-debugger
   : ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 4 | // oxlint-disable-next-line no-debugger -- some description
   `----

Found 0 warnings and 2 errors.
Finished in Xms on 1 file with 96 rules using X threads.
```

Let me know if this is an issue - we may add this natively at a future point.

Thanks!

**Comment by JasonKaz** [comment]
Fair enough, thanks!
